indexer-app/src/cmr/indexer/data/concepts/tool.clj

```diff
@@ -14,7 +12,7 @@
 (defmethod es/parsed-concept->elastic-doc :tool
   [context concept parsed-concept]
   (let [{:keys [concept-id revision-id deleted provider-id native-id user-id
-                revision-date format extra-fields]} concept
+                revision-date format extra-fields collection-associations]} concept
```

Comment:
use tool-associations here.  for es/parsed-concept->elastic-doc :service,  use service-associations.

Reply:
Then why in collections are the associations called tag-associations
                variable-associations service-associations tool-associations generic-associations and not collection-associations?

Reply:
In indexer-app/src/cmr/indexer/services/index_service.clj 
(defmethod index-concept :default
  [context concept parsed-concept options]
  .......
          (let [tag-associations (get-tag-associations context concept)
                variable-associations (get-variable-associations context concept)
                service-associations (get-service-associations context concept)
                tool-associations (get-tool-associations context concept)

......

get-service-associations and get-tool-associations are all defmulti functions.  depending on which concept it is, it calls different functions.  So to get service associations for both collection and service, you use the same service-associations, same with tool.  You just need to define the correponding defmethod.  That's how variables works. Otherwise, there is no need for the defmulti function.

Reply:
If you call collection associations, then it only applies to collections. When you call service associations, tool associations, it can apply to services, tools and collections.

Reply:
I have made the change.